## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#a1304526-c1ff-462b-aa96-8ea69f08be8a) |
<a name="a1304526-c1ff-462b-aa96-8ea69f08be8a"></a>
### 12601-55d8fac/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s8:30235
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/16) Running aws-sdk-go tests ... done in 1 seconds
(2/16) Running aws-sdk-java tests ... done in 1 seconds
(3/16) Running aws-sdk-php tests ... done in 44 seconds
(4/16) Running aws-sdk-ruby tests ... done in 4 seconds
(5/16) Running awscli tests ... done in 2 minutes and 3 seconds
(6/16) Running healthcheck tests ... done in 0 seconds
(7/16) Running mc tests ... done in 27 seconds
(8/16) Running minio-dotnet tests ... done in 33 seconds
(9/16) Running minio-go tests ... done in 2 minutes and 3 seconds
(10/16) Running minio-java tests ... done in 1 minutes and 6 seconds
(11/16) Running minio-js tests ... done in 50 seconds
(12/16) Running minio-py tests ... FAILED in 1 minutes and 37 seconds
{
  "name": "minio-py:test_list_objects_with_1001_files",
  "status": "FAIL",
  "args": {
    "bucket_name": "minio-py-test-b128014e-80c5-4252-84eb-5ceb13d10859",
    "object_name": "8d7e1039-dfb7-4ee5-8c0c-c8d8b0d341c3_0 ~ 8d7e1039-dfb7-4ee5-8c0c-c8d8b0d341c3_1000"
  },
  "message": "Listed no of objects (1999), does not match the expected no of objects (2000)",
  "error": "Traceback (most recent call last):\n  File \"/mint/run/core/minio-py/tests.py\", line 126, in _call_test\n    func(log_entry, *args, **kwargs)\n  File \"/mint/run/core/minio-py/tests.py\", line 1243, in test_list_objects_with_1001_files\n    _test_list_objects_api(bucket_name, no_of_created_files)\n  File \"/mint/run/core/minio-py/tests.py\", line 1131, in _test_list_objects_api\n    \"expected no of objects ({})\").format(no_of_files, expected_no))\nValueError: Listed no of objects (1999), does not match the expected no of objects (2000)\n",
  "duration": 68764
}
(12/16) Running s3cmd tests ... done in 25 seconds
(13/16) Running s3select tests ... done in 4 seconds
(14/16) Running security tests ... done in 0 seconds
(15/16) Running versioning tests ... done in 3 minutes and 7 seconds

Executed 15 out of 16 tests successfully.
```
Deleting image on docker hub
Deleting image locally

--- a/cmd/format-erasure.go
+++ b/cmd/format-erasure.go
@@ -242,7 +242,7 @@ func formatErasureMigrateV1ToV2(export, version string) error {
 	if err != nil {
 		return err
 	}
-	return ioutil.WriteFile(formatPath, b, 0644)
+	return ioutil.WriteFile(formatPath, b, 0666)
 }
 
 // Migrates V2 for format.json to V3 (Flat hierarchy for multipart)
@@ -284,7 +284,7 @@ func formatErasureMigrateV2ToV3(export, version string) error {
 	if err != nil {
 		return err
 	}
-	return ioutil.WriteFile(formatPath, b, 0644)
+	return ioutil.WriteFile(formatPath, b, 0666)
 }
 
 // countErrs - count a specific error.

--- a/cmd/fs-v1-multipart.go
+++ b/cmd/fs-v1-multipart.go
@@ -241,7 +241,7 @@ func (fs *FSObjects) NewMultipartUpload(ctx context.Context, bucket, object stri
 		return "", err
 	}
 
-	if err = ioutil.WriteFile(pathJoin(uploadIDDir, fs.metaJSONFile), fsMetaBytes, 0644); err != nil {
+	if err = ioutil.WriteFile(pathJoin(uploadIDDir, fs.metaJSONFile), fsMetaBytes, 0666); err != nil {
 		logger.LogIf(ctx, err)
 		return "", err
 	}

--- a/cmd/tier-journal.go
+++ b/cmd/tier-journal.go
@@ -255,7 +255,7 @@ func (jd *tierDiskJournal) Open() error {
 	}
 
 	var err error
-	jd.file, err = os.OpenFile(jd.JournalPath(), os.O_APPEND|os.O_CREATE|os.O_WRONLY|writeMode, 0644)
+	jd.file, err = os.OpenFile(jd.JournalPath(), os.O_APPEND|os.O_CREATE|os.O_WRONLY|writeMode, 0666)
 	if err != nil {
 		return err
 	}

--- a/internal/ioutil/append-file_windows.go
+++ b/internal/ioutil/append-file_windows.go
@@ -26,13 +26,13 @@ import (
 
 // AppendFile - appends the file "src" to the file "dst"
 func AppendFile(dst string, src string, osync bool) error {
-	appendFile, err := lock.Open(dst, os.O_WRONLY|os.O_APPEND|os.O_CREATE, 0644)
+	appendFile, err := lock.Open(dst, os.O_WRONLY|os.O_APPEND|os.O_CREATE, 0666)
 	if err != nil {
 		return err
 	}
 	defer appendFile.Close()
 
-	srcFile, err := lock.Open(src, os.O_RDONLY, 0644)
+	srcFile, err := lock.Open(src, os.O_RDONLY, 0666)
 	if err != nil {
 		return err
 	}

